playwright-e2e/lib/component/dsm/table/table.ts

```diff
@@ -0,0 +1,51 @@
+import { Locator, Page } from '@playwright/test';
+import ParticipantPage from 'pages/dsm/participant-page';
+
+export class Table {
+  private readonly _participantPage: ParticipantPage = new ParticipantPage(this.page);
+
+  constructor(private readonly page: Page) {}
+
+  public async openParticipantPageAt(position: number): Promise<ParticipantPage> {
+    await this.getParticipantAt(position).click();
+    await this._participantPage.assertPageTitle();
+    return this._participantPage;
+  }
+
+  public async openParticipantPageBy(columnName: string, columnValue: string, position = 1): Promise<ParticipantPage> {
+    await this.page
+      .locator(this.getParticipantXPathBy(columnName, columnValue))
+      .nth(position - 1)
+      .click();
+    return this._participantPage;
+  }
+
+  public async getParticipantDataAtBy(position: number, columnName: string): Promise<string> {
+    return await this.page.locator(this.getParticipantDataXPathAtBy(position, columnName)).innerText();
+  }
+
+  public async selectParticipantAt(position: number): Promise<void> {
+    return await this.getParticipantAt(position).nth(0).locator('mat-checkbox').click();
+  }
+
+  private getParticipantAt(position: number): Locator {
+    return this.page.locator(`//table/tbody/tr`).nth(position - 1);
+  }
+
+  /* XPaths */
+  private getParticipantXPathBy(columnName: string, columnValue: string): string {
```

Comment:
Xpath is long. I'd break into smaller pieces to help with reading. For example, I can make this new function:
```
private theadCount(columnName: string): string {
  return count(//table/thead/` +
      `th[text()[normalize-space()='${columnName}']]/preceding-sibling::th)
}
```
Then use this function inside other functions.

Reply:
Oki, Done!